job-server/src/main/scala/spark/jobserver/JobServer.scala

```diff
@@ -166,15 +169,37 @@ object JobServer {
     webApiPF(supervisor, jobInfo).start()
   }
 
-  private def joinAkkaCluster(cluster: Cluster, seedNodes: List[ActorRef]) {
-    if (seedNodes.isEmpty) {
+  /**
+    * This function is responsible for joining the Akka cluster dynamically
+    * based on current situation of Akka cluster.
+    *
+    * - If there are no old slave nodes and HA section of config is not
```

Comment:
By slave nodes, did you mean peer master nodes? I am confused by the terminology slave here.

Reply:
Sorry for confusion, by slaves, I mean driver JVMs.

This case just means that if HA is not configured and no driver jvms are alive then the current jobserver node will join itself.